central/metrics/central.go

```diff
@@ -191,17 +191,121 @@ var (
 		Name:      "pipeline_panics",
 		Help:      "A counter that tracks the number of panics that have occurred in the processing pipelines",
 	}, []string{"resource"})
+
 	sensorConnectedCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Namespace: metrics.PrometheusNamespace,
 		Subsystem: metrics.CentralSubsystem.String(),
 		Name:      "sensor_connected",
 	}, []string{"ClusterID", "connection_state"})
+
+	grpcMaxMessageSize = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.CentralSubsystem.String(),
+		Name:      "max_message_size_sent",
+		Help:      "A gauge for maximum message size seen when sending messages to sensor",
+	}, []string{"Type"})
+
+	grpcSentSize = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.CentralSubsystem.String(),
+		Name:      "grpc_message_size_sent",
+		Help:      "Histogram of sent message sizes from Central",
+		Buckets: []float64{
+			1,
+			10,
+			50,
+			100,
+			1000,
+			10_000,
+			50_000,
+			100_000,
+			500_000,
+			1_000_000,
+			2_000_000,
+			3_000_000,
+			4_000_000,
+			5_000_000,
+			6_000_000,
+			8_000_000,
+			10_000_000,
+			12_000_000,
+			15_000_000,
+			20_000_000,
+			30_000_000,
+			40_000_000,
```

Comment:
I would add 2-3 some coarser ranges: 64MB, 128MB, 256MB (the new default for the limit)